custom/icds_reports/management/commands/sql_scripts/extract_valid_phone_count_grouped_by_district.sql

```diff
@@ -0,0 +1,108 @@
+        WHEN ccs_record.add is not null AND ccs_record.add - now() < '182 days' :: interval 
+            THEN 1 
+        ELSE 0
+    END) "lm_count",
+SUM(
+    CASE 
+        WHEN ccs_record.add is not null AND ccs_record.add - now() < '182 days':: interval 
+        AND phone_number similar to '91[6789][0-9]{9}' THEN 1 else 0
+    END) "lm_valid_phone_count",
+SUM(
+    CASE
+        WHEN ccs_record.add is not null AND ccs_record.add - now() < '182 days' :: interval
+        AND (phone_number = '') IS NOT FALSE  THEN 1 ELSE 0
+    END) "lm_phone_not_available",
+SUM(
+    CASE WHEN ccs_record.add is not null AND ccs_record.add - now() < '182 days' :: interval 
+    AND (phone_number = '') IS FALSE AND phone_number NOT SIMILAR to '91[6789][0-9]{9}' THEN 1 ELSE 0
+    END) "lm_invalid_phone_count",
+
+
+-- Children
+SUM(
+    CASE 
+        WHEN child_health.dob is NOT null AND now() - child_health.dob <= '2191.5 days' ::interval  
+            THEN 1 
+        ELSE 0
+    END) "children_count",
+SUM(
+    CASE 
+        WHEN child_health.dob is NOT null AND now() - child_health.dob <= '2191.5 days' ::interval  
+        AND phone_number similar to '91[6789][0-9]{9}' THEN 1 else 0
+    END) "children_valid_phone_count",
+SUM(
+    CASE
+        WHEN child_health.dob is NOT null AND now() - child_health.dob <= '2191.5 days' ::interval
+        AND (phone_number = '') IS NOT FALSE  THEN 1 ELSE 0
+    END) "children_phone_not_available",
+SUM(
+    CASE WHEN child_health.dob is NOT null AND now() - child_health.dob <= '2191.5 days' ::interval
+    AND (phone_number = '') IS FALSE AND phone_number NOT SIMILAR to '91[6789][0-9]{9}' THEN 1 ELSE 0
+    END) "children_invalid_phone_count",
+
+-- Women 
+SUM(
+    CASE 
+        WHEN person_cases.sex = 'f' AND now() - person_cases.dob >= '4017.75 days' ::interval 
+            THEN 1 
+        ELSE 0
+    END) "women_count",
+SUM(
+    CASE 
+        WHEN person_cases.sex = 'f' AND now() - person_cases.dob >= '4017.75 days' ::interval 
+        AND phone_number similar to '91[6789][0-9]{9}' THEN 1 else 0
+    END) "women_valid_phone_count",
+SUM(
+    CASE
+        WHEN person_cases.sex = 'f' AND now() - person_cases.dob >= '4017.75 days' ::interval
+        AND (phone_number = '') IS NOT FALSE  THEN 1 ELSE 0
+    END) "women_phone_not_available",
+SUM(
+    CASE WHEN person_cases.sex = 'f' AND now() - person_cases.dob >= '4017.75 days' ::interval 
+    AND (phone_number = '') IS FALSE AND phone_number NOT SIMILAR to '91[6789][0-9]{9}' THEN 1 ELSE 0
+    END) "women_invalid_phone_count"
+
+FROM "ucr_icds-cas_static-person_cases_v3_2ae0879a" person_cases
+
+LEFT OUTER JOIN "ucr_icds-cas_static-child_health_cases_a46c129f" child_health
+
+ON 
+    child_health.mother_id = person_cases.doc_id 
+    AND 
+    child_health.state_id = person_cases.state_id 
+    AND 
+    child_health.supervisor_id = person_cases.supervisor_id
+LEFT OUTER JOIN "ucr_icds-cas_static-ccs_record_cases_cedcca39" ccs_record 
```

Comment:
Instead of doing both the joins in same query, it's good to have a different query for both ccs and child

Reply:
Also, person_case is not left joined to ccs case or child case, child case or ccs case is left joined to person case, afaik from the dashboard perspective.

Reply:
> Instead of doing both the joins in the same query, it's good to have a different query for both ccs and child
I was thinking of getting the entire output in one query, If it will make big difference it can be broken down.

> person_case is not left joined to ccs case or child case, child case or ccs case is left joined to person case
I did not quite understand this statement, I may be wrong on this but if I am extracting information from person case itself then I am supposed to left join person case with others?